### DISCUSSION TITLE
Could this framework be used to create a mod that enables haptic feedback/adaptive trigger?

### DISCUSSION THREAD

**Discussion by DamianPereira** [praydog/REFramework#838] [General Questions]
Hi! I was wondering if reframework can hook into input events and trigger arbitrary code (like calls to dualsense's adaptive trigger). I'm trying to understand the docs, but I don't want to get myself too deep into it if someone can tell me it's not possible.

**Comment by praydog** [comment]
Yes. If you want to trigger PS5 code then you'll need to use a native C++ plugin to do so, you cannot do this from Lua unless you expose a function to Lua to do so. Once that is done, you can find the input code in the game and hook it.

**Reply by DamianPereira** [reply]
Thank you! And Do you think I can make haptic feedback also be triggered in a similar way? I think I'd need to play audio files to a specific device, when certain actions happen. As far as I understand haptic feedback is audio that is sent to the vibration motors. Once I'm running C++ code I should be able to interface with DualsenseX though.

I'll explore making a mod for it. Thanks for confirming this framework can be used.
